.travis.yml

```diff
@@ -1,9 +1,49 @@
 language: python
 sudo: false
+env:
+  # Downloadable toolchains & SDKs for esp8266 & esp32
+  #
+  # Toolchains used for automated builds:
+  # ESP8266 - prebuilt gcc 4.8.2 from ESP8266 Arduino
+  # ESP32 - official Espressif toolchain
+  #
+  # Note: if these are updated, you need to flush the Travis cache
+  #
+  # (esp-idf is downloaded a zip file. It doesn't fully work this way, but
+  # we're just using it for some header files.)
+
+  # Toolchains
+  ESP8266_TOOLCHAIN_URL="http://arduino.esp8266.com/linux64-xtensa-lx106-elf.tar.gz"
```

Comment:
oh, i didn't realize i was commenting on the `.travis.yml` file, so ignore that!